jobs/e2e-test-cypress.groovy

```diff
@@ -92,28 +96,42 @@ def runLamdaTest() {
       get qvYpo_KnpCiLBN69fYpEYA --format json | jq -r .password\
       """, returnStdout:true).trim();
    
-   // TODO(ruslan): implement TEST_TYPE param to use decorator or flag 
+   // TODO(ruslan): Implement TEST_TYPE param to use decorator or flag 
    // in cypress script files. 
+   // TODO(ruslan): Use build tags --bt with prod/znd states.
    def runLambdaTestArgs = ["yarn",
                             "lambdatest",
                             "--cy=\"--config baseUrl=${params.URL}\", retries=${params.TEST_RETRIES}",
                             "--bn",
-                            "e2e-test # (${params.URL}: ${params.REVISION_DESCRIPTION}) @${params.DEPLOYER_USERNAME}",
+                            BUILD_NAME,
                             "-p",
-                            "${params.NUM_WORKER_MACHINES}"
+                            "${params.NUM_WORKER_MACHINES}",
+                            "--sync=true", 
+                            "--bt",
+                            "e2e-test #${env.BUILD_NUMBER}",
+                            "--eof"
    ];
    
    dir('webapp/services/static') {
       withEnv(["LT_USERNAME=${lt_username}",
                "LT_ACCESS_KEY=${lt_access_key}"]) {
-         exec(runLambdaTestArgs)   
+         exec(runLambdaTestArgs); 
       }
    }
 }
 
+
 onWorker("ka-test-ec2", '6h') {
+   // TODO(ruslan): Remove @ in extraText before merging with current e2e-test pipeline, 
+   // we don't want to disturb real users.  
    notify([slack: [channel: params.SLACK_CHANNEL,
-                  when: ['FAILURE', 'UNSTABLE']]]) {
+                   sender: 'Testing Turtle',
+                   emoji: ':turtle:',
+                   extraText : 'Hey '+'@' + 
+                   params.DEPLOYER_USERNAME + 
+                   " BUILD: " + BUILD_NAME + 
+                   ') FAILED (<https://automation.lambdatest.com/logs/|Open>)',
```

Comment:
Doesn't notify.groovy already say that the build failed?  I'm worried this is just going to repeat information that's already there, and the only new thing will be the lambdatest link.  But we can always tweak this text later.

It would be awesome, btw, the if the link could go directly to the logs for this run, rather than the generic "logs" page.  Is that posible?

Reply:
Yeah, `notify.groovy` part is responsible for the first chunk like: 
```
deploy/e2e-test-cypress failed #59 ([Open](https://jenkins.khanacademy.org/job/deploy/job/e2e-test-cypress/59/))
deploy/e2e-test-cypress failed: See https://jenkins.khanacademy.org/job/deploy/job/e2e-test-cypress/59/flowGraphTable/ for what failed (look for the red circle(s)).
```
And with the rest block, we might want to ping the user directly. The only similar custom `_alert` function I found in `deply-webapp.groovy` 

Concerning putting the link directly to the logs for this run, yeah it was working on the previous PR version:
```
BUILD_ID = sh(
      script: "echo '$response' | jq -r '.data | map(select(.name ==" +
              " \"${BUILD_NAME}\"))[0].build_id'", 
      returnStdout: true).trim();
```  
Let me try a few tweaks on how to implement this API to the current version. The tricky part is we have to first execute `lambda-cli` and after do a call to get freshly generated `build_id` that we can use in the link.

Reply:
Ok, I was able to get that BUILD_ID with `60s` delay parallel job in `run-notify`: 
```
parallel([
         "run-lambda": { exec(runLambdaTestArgs); },
         "run-notify": { notifyFailures(); }, // wait for ~60 sec to have generated lambdatest build_id 
])
```
But the problem is how to pass BUILD_ID in the upstream `notify.groovy` wrapper from the stage... obviously, it returns null?

Not sure, but looks like we have to create some [custom notify function](https://github.com/Khan/jenkins-jobs/pull/47/commits/491a18830186f0331ff5d545baaa7c6f2a2be926#diff-34a902da56226c29ae9f839e49290d303ad18000e36ea6096698afcbbf48be63R145) to have this lambda logs link?
@csilvers WDYT?

Reply:
I'd say: just put in a TODO for now.  We can iterate on it over time, but let's get live what you have now.

Reply:
Sounds good!